eng/pipelines/templates/scripts/mashup-doc-index.yml

```diff
@@ -0,0 +1,85 @@
+parameters: 
+  TargetFolder: ''
+
+steps:
+  - task: PythonScript@0
+    displayName: MashUp Generated Index Site so its served from default site location
+    inputs:
+      scriptSource: inline
+      script: |
+        import argparse
+        import os
+        import logging
+        import re
+        import shutil
+        from io import open
+
+        SITE_INDEX = r'${{ parameters.SourceDirectory }}\docfx_project\_site'
+        TOC_HTML_REGEX = r"\.\./toc.html"
+        NAV_TOC_HTML_REGEX = r"api/"
+        PREV_DIR_REGEX = r"\.\./"
```

Comment:
This assumes that any and all `one directory up` indicators must be replaced. Given that this is generating an index, and no contents of packages (like readmes, other than the the main readme), is this a safe assumption?

Aka, This won't replace any occurrences of `..` in a code snippet?

Reply:
Yup I believe it is. The only other things in the code are directory names which don't contain slashes. There should be no case where there is an exception as long as this is used for just the index.